src/dc_configure.rs

```diff
@@ -13,6 +13,22 @@ use crate::param::Params;
 use crate::types::*;
 use crate::x::*;
 
+macro_rules! progress {
```

Comment:
Is there a reason to use a macro over a method?

Reply:
yeah, I hope the compiler optimizes it(if statement).
Also why does rust use a macro for printing?
I would have used a method, but I dind't find a range type to replace the if statement.
PS: the C code uses also a "macro".

Reply:
Would like to get some rust pros opinion on this.
@dignifiedquire @link2xt @KAction any ideas?